### PR TITLE
Install router

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do?
- Installs React Router as a dev dependency
- Wraps App component in BrowserRouter component
- Adds Routes to home view and movie details card
- Adds Links to movie posters that lead to corresponding movie details cards
- Wraps back button in Link component to return to home view

## How should it be tested?
 - All functionality should be there!
 - Click on movie poster, see movie details, hit X to exit, hit forward and backward buttons in browser to see changes!
 
*** Movie detail cards lost some info, needs to be fixed!

### CODE REVIEW

**Comment by m1073496** [comment]
[Iteration 4: URL Updates](https://trello.com/c/fEDktckG/21-iteration-4-url-updates)

**Comment by m1073496** [comment]
[Iteration 4: URL Navigation](https://trello.com/c/eEg9MA0e/22-iteration-4-url-navigation)

File: src/app/App.js [hunk 1]
```diff
@@ -19,9 +20,19 @@ class App extends Component {
     return (
       <main className="App">
         <h1>Rancid Tomatillos</h1>
+        <Route exact path="/" render={() => <MovieContainer movies={this.state.movies} displayMovieDetails={this.displayMovieDetails} /> } />
+        <Route
+            path="/:id"
+            render={({ match }) => {
+              if (this.state.movies.length > 0) {
+                const { id } = match.params
```
Comment (cbmackintosh) [inline src/app/App.js:28]: Good solution on line 27. I was wondering what would happen if we navigated straight to a movie details card, or clicked the refresh button on a movie details card. The card wouldn't be able to render without App creating the all movies array in state first. But this conditional accounts for that. We can probably shorten it to just (this.state.movies.length)

File: src/details/MovieDetailsCard.js [hunk 2]
```diff
@@ -23,13 +24,15 @@ class MovieDetailsCard extends Component {
       return (
         <div className='movie-details-card' style={{backgroundImage: `url(${this.state.movie.backdrop_path})`}}>
          <div className='movie-details-text'>
-           <button className='close-button' onClick={() => this.state.displayMovieLibrary()}>X</button>
+          <Link to={`/`}>
+            <button className='close-button'>X</button>
+          </Link>
            <h1 className='movie-title'>{this.state.movie.title}</h1>
```
Comment (cbmackintosh) [inline src/details/MovieDetailsCard.js:30]: I think moving the className='close-button' up to the Link element will solve the styling issue we're seeing here. Might impact the tests though...

File: src/details/MovieDetailsCard.js [hunk 3]
```diff
@@ -41,12 +44,12 @@ class MovieDetailsCard extends Component {
...
-            </tbody> 
-           </table>
+            </tbody>
+           </table>*/}
         </div>
        </div>
       )
-    }   
+    }
   }
```
Comment (cbmackintosh) [inline src/details/MovieDetailsCard.js:54]: I'd like to move the movie details network request from App into the Movie Details Card, ie give MovieDetailsCard its own componentDidMount method. I think that will help us render all that additional information from the server.

File: src/poster/MoviePoster.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
-import React from 'react'
+import React from 'react';
+import { Link } from 'react-router-dom';
 import './MoviePoster.css'
 
 const MoviePoster = ({id, poster_path, title, displayMovieDetails}) => {
   return (
     <div id={id} className='movie-poster' onClick={() => displayMovieDetails(id)}>
```
Comment (cbmackintosh) [inline src/poster/MoviePoster.js:7]: I think we can get rid of that onClick property now that we have the Link set up?

**Review by cbmackintosh — APPROVED** [review]
Nice work! This lays the groundwork for setting up Router in the application. We just need to refine it to get those additional movie details to populate on the movie details page without running into any errors.

**Pull request merged**